docs/spec/spec.md

```diff
@@ -0,0 +1,60 @@
+# Specification: Ballerina Regex Library
+
+_Owners_: @daneshk @kalaiyarasiganeshalingam  
+_Reviewers_: @daneshk  
+_Created_: 2021/12/07   
+_Updated_: 2021/12/07  
+_Issue_: [#2323](https://github.com/ballerina-platform/ballerina-standard-library/issues/2323)
+
+# Introduction
+This is the specification for the Regex standard library, which provides functionalities such as matching, 
+replacing and splitting strings based on regular expressions in the 
+[Ballerina programming language](https://ballerina.io/), which is an open-source programming language for the cloud that
+makes it easier to use, combine, and create network services.
+
+# Contents
+1. [Overview](#1-overview)
+2. [Operations](#2-operations)
+    * 2.1. [Matches](#21-matches)
+    * 2.2. [Replace All](#22-replace-all)
+    * 2.3. [Replace First](#23-replace-first)
+    * 2.4. [Split](#24-split)
+
+# 1. Overview
+This library is based on [regular expressions](https://en.wikipedia.org/wiki/Regular_expression), which are notations 
+for describing sets of character strings that specify a search pattern.
+
+# 2. Operations
+
+## 2.1. Matches
+This is used to check whether a string matches the provided regex.
+```ballerina
+boolean isMatched = regex:matches("Ballerina is great", "Ba[a-z ]+");
```

Comment:
Shall we mention add API definition instead of a sample

Reply:
like https://github.com/ballerina-platform/module-ballerina-time/blob/master/docs/spec/spec.md

Reply:
Fixed in c823587